src/linker/implementation.py

```diff
@@ -0,0 +1,37 @@
+from pathlib import Path
+
+from linker.step import Step
+from linker.utilities.general_utils import load_yaml
+
+
+class Implementation:
+    def __init__(self, step: Step, directory: Path):
+        self.step = step
+        self.directory = directory
+        self.name = directory.name
+        self.container_full_stem = self._get_container_full_stem()
+
+    def run(self, runner, container_engine, input_data, results_dir):
```

Comment:
Runner is a callable? Does that mean you could pass a different callable?

Reply:
Yes but it specifically gets set in `runner.main` - it's not intended to be set by a user though it indeed could be